src/inmanta/util.py

```diff
@@ -747,3 +748,38 @@ def ensure_event_loop() -> asyncio.AbstractEventLoop:
         new_loop = asyncio.new_event_loop()
         asyncio.set_event_loop(new_loop)
         return new_loop
+
+
+class ExhaustedPoolWatcher:
```

Comment:
Let's not make this a singleton. We also have a connection pool on the lsm extension. So we will need two instances of this class in the end.